pkg/apis/nmstate/v1alpha1/nodenetworkconfigurationenactment_types.go

```diff
@@ -0,0 +1,50 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeNetworkConfigurationEnactmentList contains a list of NodeNetworkConfigurationEnactment
+type NodeNetworkConfigurationEnactmentList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []NodeNetworkConfigurationEnactment `json:"items"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeNetworkConfigurationEnactment is the Schema for the nodenetworkconfigurationenactments API
+// +k8s:openapi-gen=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=nodenetworkconfigurationenactments,shortName=nnce,scope=Cluster
+type NodeNetworkConfigurationEnactment struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Status NodeNetworkConfigurationEnactmentStatus `json:"status,omitempty"`
+}
+
+// NodeNetworkConfigurationEnactmentStatus defines the observed state of NodeNetworkConfigurationEnactment
+// +k8s:openapi-gen=true
+type NodeNetworkConfigurationEnactmentStatus struct {
```

Comment:
NNCE is not completed until we have rendered state as part of the status. It can be done in a follow-up.

Reply:
Let's do it here, so we have a complete CRD, the content is just about copy desiredState right now.